core/controllers/editor.py

```diff
@@ -563,13 +563,11 @@ def get(self, exploration_id, escaped_state_name):
 
                 trained_answers = set()
                 for answer_group in interaction.answer_groups:
-                    for rule_spec in answer_group.rule_specs:
-                        if (rule_spec.rule_type ==
-                                exp_domain.RULE_TYPE_CLASSIFIER):
-                            trained_answers.update(
-                                interaction_instance.normalize_answer(trained)
-                                for trained
-                                in rule_spec.inputs['training_data'])
+                    if answer_group['training_data']:
```

Comment:
Why do you need the "if" check (i.e. what is it guarding against)? 

If you're guarding against an empty list -- I don't think you actually need the check?

Reply:
We do need check because otherwise it will be shown in training data as a class without any training instances and so the classifier algorithm might get worse accuracy if we have such empty classes. Although the effect is different for the different classifiers, removing such classes is always a good thing as a data cleaning step.

Reply:
Sorry, not sure I understand. If you were to remove the check, then in the case where answer_group['training_data'] is an empty list, you would be doing trained_answers.update([]) which is a no-op. Am I missing something?

Reply:
Oops, my bad. I thought this is the function which generates training data for mapping job but it is something else. Yeah, we can remove this.

Reply:
Done